docs/gdd.rst

```diff
@@ -0,0 +1,272 @@
+This game utilizes a "job system."
+Each monster has access to a set of jobs and has a currently active job.
+Different jobs can provide different stat boosts and access to certain abilities.
+Jobs can also have requirements that must be met before they can be used (e.g., a certain number of levels in another job).
+
+A monster gains "experience" in a job via "Job Points" (or JP).
+After acquiring enough JP, a monster will obtain a new level in the current job.
+This is no limit to the number of levels that can be acquired for a job.
+However, job levels have no benefits themselves (e.g., stat gains).
+Instead, job levels can used to unlock new classes, and the JP earned can be spent on purchasing new :ref:`abilities`.
+Jobs can also have a number of stat boosts available for purchase with JP.
+
+Details on all available jobs can be found :doc:`here <_gen/jobs>`.
+
+Tags
+^^^^
+
+Various requirements in the game are determined by "tags."
+Some examples of requirements include:
+
+* what jobs a Golem has access to
+* what breeds a trainer has access to
+
+Monsters and trainers each have a pool of tags.
+To determine eligibility the item being checked for (e.g., a job) is compared to the appropriate tag pool.
+If all tags on the item are present in the tag pool, the requirements are met.
+In other words, the required tags are an "all of" relationship.
+
+A trainer's tag pool contains:
+
+* Any "personal tags" acquired
+
+A monster's tag pool contains:
+
+* The trainer's tags
+* The breed's tags
+* A tag for each item in a Cartesian product of available jobs and their levels.
+  For example, a monster with three levels in :ref:`job-squire` and two levels in :ref:`job-ruinsmith` will have the following tags:
+
+   * ``job_squire_1``
+   * ``job_squire_2``
+   * ``job_squire_3``
+   * ``job_ruinsmith_1``
+   * ``job_ruinsmith_2``
+
+Monster Stats
+^^^^^^^^^^^^^
+
+The monster stats are:
+
+Hit Points
+   The amount of damage a monster can take before going down
+
+Magic Points
+   A resource for using abilities
+
+Physical Attack
+   Multiplier for damage on ``physical`` abilities
+
+Magical Attack
+   Multiplier for damage on ``magical`` abilities
+
+Movement
+   The number of tiles the monster can move in a single turn
+
+.. _abilities:
+
+Abilities
+^^^^^^^^^
+
+Monsters have access to a set of abilities.
+By default, all monsters have a "Basic Attack" as determined by their active job.
+In addition to this "Basic Attack," monsters have purchased abilities.
+Abilities can be purchased for the active job with JP earned while that job has been active.
+Different jobs have access to different abilities.
+
+Once an ability has been purchased, the monster will continue to have access to it even when changing jobs.
+This can lead to some fun and interesting monster builds.
+
+Abilities have the following stats:
+
+MP Cost
+   How much MP is required to use this ability (can be zero)
+
+JP Cost
+   How much JP is required to purchase this ability
+
+Type
```

Comment:
Is the type the damage type?

Can this be both Physical and Magical?

Is the power based on the job or does each ability have it's own type different than the parent job?

If based on the job, perhaps instead it couples to the job itself and pulls the value from that?

Reply:
Yes, this is the damage type.
No, this cannot (currently, it's in additional ideas) be both physical and magical.
Jobs have no inherent damage type; only abilities have a type.

Reply:
Sounds good!